lact-gui/src/app/graphs_window/plot/render_thread.rs

```diff
@@ -0,0 +1,362 @@
+use super::cubic_spline::cubic_spline_interpolation;
+use super::to_texture_ext::ToTextureExt;
+use super::PlotData;
+use anyhow::Context;
+use cairo::{Context as CairoContext, ImageSurface};
+
+use gtk::gdk::MemoryTexture;
+use itertools::Itertools;
+use plotters::prelude::*;
+use plotters::style::colors::full_palette::DEEPORANGE_100;
+use plotters_cairo::CairoBackend;
+use std::cmp::max;
+use std::ops::{Deref, DerefMut};
+use std::sync::{Arc, Mutex};
+use thread_priority::{ThreadBuilderExt, ThreadPriority};
+use tracing::error;
+
+enum Request {
+    Terminate,
+    Render(RenderRequest),
+}
+
+#[derive(Default)]
+pub struct RenderRequest {
+    pub title: String,
+    pub value_suffix: String,
+    pub secondary_value_suffix: String,
+    pub y_label_area_size: u32,
+    pub secondary_y_label_area_size: u32,
+
+    pub data: PlotData,
+
+    pub width: u32,
+    pub height: u32,
+
+    pub supersample_factor: u32,
+}
+
+#[derive(Default)]
+struct RenderThreadState {
+    request_condition_variable: std::sync::Condvar,
+    last_texture: Mutex<Option<MemoryTexture>>,
+    current_request: Mutex<Option<Request>>,
```

Comment:
Is  `Condvar` + `current_request` behind a mutex somehow different from using a standard mpsc channel? It seems to be used exactly like a channel

Reply:
Tokio's `notify` channel type has this behaviour, but it doesn't have having blocking versions of the calls like the other channels (only async), so it won't work here. I suppose manual Condvar is fine